torchgeo/models/cnnlstm.py

```diff
@@ -0,0 +1,378 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""CNN-LSTM Model Architecture."""
+
+from typing import Any
+
+import timm
+import torch
+import torch.nn as nn
+
+
+class CNN(nn.Module):
+    """CNN backbone using a TIMM model.
+
+    Args:
+        model_name (str): Name of the TIMM model architecture to use.
+        pretrained (bool): Whether to load pretrained weights.
+        output_dim (int, optional): Desired output feature dimension. If None, uses
+            the TIMM model's native feature dimension.
+        freeze_backbone (bool): If True, freezes the backbone parameters.
+        in_chans (int): Number of input channels (e.g., 3 for RGB).
+
+    Raises:
+        ImportError: If TIMM library is not available.
+
+    Forward:
+        x (torch.Tensor): Input tensor of shape either
+            - (B, T, C, H, W) for sequences of images, or
+            - (B, C, H, W) for single images.
+
+    Returns:
+        torch.Tensor: Extracted features of shape (B, T, output_dim) or (B, output_dim).
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        pretrained: bool = True,
+        output_dim: int | None = None,
+        freeze_backbone: bool = False,
+        in_chans: int = 3,
+    ) -> None:
+        """Initialize CNN backbone with TIMM model.
+
+        Args:
+            model_name: Name of the TIMM model architecture to use.
+            pretrained: Whether to load pretrained weights.
```

Comment:
We use `weights` for other models to match torchvision's newer weights API. It should take a Weights enum value as input